master/docs/developer/style.rst

```diff
@@ -122,6 +123,15 @@ Deferred.  As a shortcut, ``d.addCallback`` works as a decorator::
     def rev_parse(_): # note use of '_' to quietly indicate an ignored parameter
         return utils.getProcessOutput(git, [ 'rev-parse', revname ])
 
+.. note::
+
+    ``d.addCallback`` is not really a decorator as it does not return a modified function.
+    As a result in previous code, ``rev_parse`` value is actually the Deferred.
+    In general the :class:`inlineCallbacks` method is preferred inside new code as it keeps the code easier to read.
+    As a general rule of thumb, when you need more than 2 callbacks in the same method, its time to switch it to  :class:`inlineCallbacks`.
+    This would be for example the case for previous :py:func:`getRevInfo`.
```

Comment:
❓ And as i understand it, this would attempt to link to the Python docs....should probably just be ``getRevInfo``